xesmf/frontend.py

```diff
@@ -555,7 +559,11 @@ def regrid_dataset(self, ds_in, keep_attrs=False, skipna=False, na_thres=1.0):
         ]
         ds_in = ds_in.drop_vars(non_regriddable)
 
-        ds_dtypes = [d.dtype for d in ds_in.data_vars.values()]
+        out_dtype = (  # Useful for dask-backed datasets
```

Comment:
wouldn't calling `.values()` trigger the loading into memory of the array??

I also think we could make this more generic with something like:

```python
dtypes = list(set(ds_in.data_vars.dtype))
if len(dtypes) == 1:
    out_dtype = dtypes[0]
else:
    size = 1  # init to 1 byte
    for d in dtypes:
        size = max(size, d.itemsize)  # use max number of bytes found in dtypes
    out_dtype = np.dtype(f"f{size}")
```